src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -0,0 +1,391 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Linq;
+using ProtoCore.AssociativeGraph;
+using ProtoCore.AssociativeEngine;
+using ProtoCore.AST;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.BuildData;
+using ProtoCore.CodeModel;
+using ProtoCore.DebugServices;
+using ProtoCore.DSASM;
+using ProtoCore.Lang;
+using ProtoCore.Lang.Replication;
+using ProtoCore.Runtime;
+using ProtoCore.Utils;
+using ProtoFFI;
+
+using StackFrame = ProtoCore.DSASM.StackFrame;
+
+namespace ProtoCore
+{
+    /// <summary>
+    /// The RuntimeData is an object that contains properties that is consumed only by the runtime VM
+    /// It is instantiated prior to execution and is populated with information gathered from the CompileCore
+    /// 
+    /// The runtime VM is designed to run independently from the front-end (UI, compiler) 
+    /// and the only 2 properties it needs are the RuntimeData and the DSExecutable.
+    /// 
+    /// The RuntimeData will also contain properties that are populated at runtime and consumed at runtime.
+    /// </summary>
+    public class RuntimeData
+    {
+
+        #region COMPILER_GENERATED_READ_ONLY
```

Comment:
Yes agree, will follow this up with all intended read-only in the next immediate PR